File: compiled_lang/go/src/e60.branch.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"regexp"
)

func main() {
	fmt.Print("Input a character: ")
	reader := bufio.NewReader(os.Stdin)
	b, _ := reader.ReadByte()
	s := string(b)

	if regexp.MustCompile("[A-Z]").MatchString(s) {
		fmt.Println("Uppercase letter")
	} else if regexp.MustCompile("[a-z]").MatchString(s) {
		fmt.Println("Lowercase letter")
	} else if regexp.MustCompile("[0-9]").MatchString(s) {
		fmt.Println("Digit")
	} else {
		fmt.Println("Punctuation, whitespace, or other")
	}
}

File: lessons/gen_scripts/php/scripts/e60.branch.php
#!/usr/bin/env php
<?php
echo "Input a character: "; $Keypress = fgetc(STDIN);
 
if (preg_match("/[a-z]/", $Keypress)) {
    echo "Lowercase letter\n";
} elseif (preg_match("/[A-Z]/", $Keypress)) {
    echo "Uppercase letter\n";
} elseif (preg_match("/[0-9]/", $Keypress)) {
    echo "Digit\n";
} else {
    echo "Punctuation, whitespace, or other\n";
}
?>
